ui_framework/src/components/context_menu/context_menu.js

```diff
@@ -27,55 +29,95 @@ export class KuiContextMenu extends Component {
     super(props);
 
     this.resetTransitionTimeout = undefined;
+    this.menuItems = [];
 
     this.state = {
       outGoingPanelId: undefined,
       currentPanelId: props.initialPanelId,
       transitionDirection: undefined,
+      isTransitioning: false,
+      focusedMenuItemIndex: 0,
     };
   }
 
+  onKeyDown = e => {
+    switch (e.keyCode) {
+      case cascadingMenuKeyCodes.UP:
+        if (this.menuItems.length) {
+          e.preventDefault();
+          const nextFocusedMenuItemIndex = this.state.focusedMenuItemIndex - 1;
+
+          this.setState({
+            focusedMenuItemIndex: nextFocusedMenuItemIndex < 0 ? this.menuItems.length - 1 : nextFocusedMenuItemIndex,
```

Comment:
Since we are updating the state from a previous state, we should rather use [`setState`](https://facebook.github.io/react/docs/react-component.html#setstate) with an updater function here, to avoid issues with the async update. (This might even cause the issue @snide were having.)